fml/trace_event.cc

```diff
@@ -22,6 +23,21 @@ namespace tracing {
 
 #if TIMELINE_ENABLED
 
+static AsciiTrie s_whitelist;
+
+// This is a macro to force inlining to help unoptimized code speed.
```

Comment:
This is not a concern we optimize for. Please avoid macros and use functions instead.

Reply:
Typically I'd agree with you but tracing is a special case.  This code is stripped out in optimized builds, we mostly care about the performance of debug builds which doesn't get optimized.  The nature of tracing is that it provides overhead across everything.  That's why I took this extraordinary measure.  I wanted Debug builds who don't supply 'trace-whitelist` to get the smallest hit possible.

Reply:
I think you are conflating debug (the Flutter runtime mode) with optimized (compiler optimization controlled with -On and NDEBUG). This code will NOT be stripped in optimized builds but will be optimized in debug builds. Your concerns are valid for debug_unopt builds in which tracing is irrelevant anyway (I'd argue tracing is only valid in profile but thats besides the point).

Reply:
Guidance on macros in the style guide: go/fluttercpp#Preprocessor_Macros

Reply:
Good point, tracing is really most useful for profile builds, but debug_unopt build still have to deal with the cost =T